R/new-data-frame.r

```diff
@@ -0,0 +1,22 @@
+# Fast data.frame constructor
+# No checking, recycling etc. unless asked for
+new_data_frame <- function(..., .check = FALSE) {
```

Comment:
Can you make this match the conventions used elsewhere? i.e. `new_data_frame()` should take a list and an `n` argument, and instead of a check argument, should provide a `validate_data_frame()` function.

See <https://github.com/r-lib/vctrs/blob/master/R/type-data-frame.R#L18> for my most recent attempt at a constructor (but you don't need to worry about subclassing here)

Reply:
You might want to skim https://adv-r.hadley.nz/s3.html to get my latest advice on S3 constructors etc.

Reply:
Just to get you right - you want the constructor to have a validation argument that takes a user specified function?